corehq/apps/events/forms.py

```diff
@@ -104,6 +111,38 @@ def __init__(self, *args, **kwargs):
         self.fields['expected_attendees'].choices = self.get_attendee_choices()
         self.fields['attendance_takers'].choices = self._get_possible_attendance_takers_ids()
 
+        self.fields['name'].disabled = not fields_should_be_available['name']
+        self.fields['start_date'].disabled = not fields_should_be_available['start_date']
+        self.fields['attendance_target'].disabled = not fields_should_be_available['attendance_target']
+        self.fields['expected_attendees'].disabled = not fields_should_be_available['expected_attendees']
+        self.fields['attendance_takers'].disabled = not fields_should_be_available['attendance_takers']
+
+    def determine_field_availability(self, event):
+        availability = {
+            'name': int(True),
+            'start_date': int(True),
+            'end_date': int(True),
+            'attendance_target': int(True),
+            'sameday_reg': int(True),
+            'tracking_option': int(True),
+            'expected_attendees': int(True),
+            'attendance_takers': int(True),
+        }
+
+        if event:
+            event_not_started = event.attendee_list_status == NOT_STARTED
```

Comment:
Nit: What if you give defaults to `event_not_started` and `event_in_progress` and then only change these `if event`? You can then completely remove most of what's currently in the conditional statement.

```
event_not_started = True
event_in_progress = False
if event:
    event_not_started = event.attendee_list_status == NOT_STARTED
    event_in_progress = event.attendee_list_status == IN_PROGRESS
return {
    'name': int(event_not_started),
    ...
}
```

Reply:
Good suggestion, thanks :+1: